### PR TITLE
Goal Criteria form redesign

### LINKED ISSUE #1961: Goal Criteria form redesign
Should similar view as goal deps from
Radiobuttons must contains Simple or Goal type
Hidden the criteria weight field

### PR DESCRIPTION
**Pull request by LamaEats**
New implementation of Criteria Form

Can be use for adding or editing criteria, also creating criteria from current goal / issue in the other goal
## PR includes

- [x] Refactor
- [x] Feature

## Related issues

Resolve #1961 

## QA Instructions, Screenshots, Recordings

Usage

```ts
interface CriteriaFormProps {
    // suggest items
    items: SuggestItem[];
    // default selected mode: simple criteria or goal in current implementation
    defaultMode?: CriteriaFormMode;
    // allow hide radio button for control form mode
    withModeSwitch?: boolean;
    // form values for edit early saved criteria
    values?: Partial<CriteriaFormValues>;
    // additional data for validate form data
    validityData: { title: string[]; sumOfCriteria: number };

    // action callbacks
    onModeChange?: (mode: CriteriaFormMode) => void;
    onSubmit: (values: CriteriaFormValues) => void;
    onCancel: () => void;
    onItemChange?: (item?: SuggestItem) => void;
    onInputChange?: (value?: string) => void;

    // render item function 
    renderItem: React.ComponentProps<typeof AutoComplete<SuggestItem>>['renderItem'];
}

<CriteriaForm
    defaultMode="goal"
    withModeSwitch
    onInputChange={onInputChange}
    onItemChange={onGoalChange}
    onSubmit={onSubmit}
    onCancel={onCancel}
    items={itemsToRender}
    validityData={{title: [], sumOfCriteria: 0}}
    renderItem={(props) => ReactNode}
/>
```
image
image
image
image
image

### CODE REVIEW

File: src/components/CriteriaFormV2/CriteriaForm.tsx [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+const StyledWeightFieldWrapper = styled.div`
+    display: inline-flex;
+    align-items: center;
+    justify-content: flex-start;
+    gap: ${gapS};
+    height: 28px; // input height
+`;
+
+const StyledText = styled(Text)`
+    white-space: nowrap;
```
Comment (awinogradov) [inline src/components/CriteriaFormV2/CriteriaForm.tsx:149]: Text has prop for that

File: src/components/CriteriaFormV2/CriteriaForm.tsx [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+    const handleSelectItem = useCallback(
+        ([item]: SuggestItem[]) => {
+            setValue('selected', item);
+            setValue('title', item.title);
+        },
+        [setValue],
+    );
+
+    const handleFormSubmit = useCallback(() => {
```
Comment (awinogradov) [inline src/components/CriteriaFormV2/CriteriaForm.tsx:298]: Do u really need wrapper?

File: src/components/CriteriaFormV2/CriteriaForm.tsx [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+
+export const maxPossibleWeight = 100;
+export const minPossibleWeight = 1;
+
+const schema = z.object({
+    mode: z.enum<CriteriaFormMode, Readonly<[CriteriaFormMode, CriteriaFormMode]>>(['simple', 'goal']),
+    weight: z.string().optional(),
+    title: z
+        .string({
+            required_error: 'Title is required',
```
Comment (awinogradov) [inline src/components/CriteriaFormV2/CriteriaForm.tsx:48]: i18n

File: src/components/CriteriaFormV2/CriteriaForm.tsx (hunk 1 above)
Comment (LamaEats) [reply src/components/CriteriaFormV2/CriteriaForm.tsx:165]: No, haven't
image

**Review by awinogradov — APPROVED** [review]

**Review by DenisVorop — CHANGES_REQUESTED** [review]
i18n won't work outside of rendering

File: src/components/CriteriaFormV2/CriteriaForm.tsx [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+    onItemChange?: (item?: SuggestItem) => void;
+    onInputChange?: (value?: string) => void;
+
+    renderItem: React.ComponentProps<typeof AutoComplete<SuggestItem>>['renderItem'];
+}
+
+const radios: Array<{ value: CriteriaFormMode; title: string }> = [
+    { title: tr('Simple'), value: 'simple' },
+    { title: tr('Goal'), value: 'goal' },
+];
```
Comment (DenisVorop) [inline src/components/CriteriaFormV2/CriteriaForm.tsx:144]: i18n won't work outside of rendering

**Review by DenisVorop — CHANGES_REQUESTED** [review]

**Review by DenisVorop — APPROVED** [review]

**Pull request merged**